app.py

```diff
@@ -355,6 +376,31 @@ def main():
 	"""Redirect to main page."""
 	return redirect(url_for('login'))
 
+@app.route('/annotate/direct_entry', methods=['GET'])
+@loginrequired
+def direct_entry():
+	"""Directly enter a sentence."""
+	sent = request.args.get('sent', '').strip()
+	sentid = str(request.args.get('id', '')).strip()
+	if len(sent) == 0:
+		return jsonify({'error': 'Sentence is empty.'})
+	elif len(sentid) == 0:
+		return jsonify({'error': 'Sentence ID is empty.'})
```

Comment:
Any other validation for the sentence ID? Is it allowed to have internal whitespace?

Reply:
I checked: internal whitespace is in the sentence ID field is OK for the database, and I think it's generally OK for this workflow. We could consider some form validation to prevent injection attacks (here and other places where users enter text that's processed on the server side).